File: Cap 05/5.3 - Laco Condicional Pre-Teste/5.3.1 - Controle Condicional Verdadeiro/Efetivo/Interativo/enquantoINT.cpp
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 5 - Topico 5.3.1
 * 
 * Programa ..: LACO CONDICIONAL PRE-TESTE COM FLUXO VERDADEIRO INTERATIVO
 * 
 */

#include <iostream>
#include <cctype>
using namespace std;

int N, R;
char RESP;
int main(void) {
  cout << "LACO PRE-TESTE COM FLUXO VERDADEIRO" << endl;
  cout << "Multiplicacao de numero por 3 por N vezes" << endl << endl;
  RESP = 'S';
  while (RESP == 'S') {
    cout << "Entre valor numerico inteiro: ";
    cin >> N;
    R = N * 3;
    cout << "O numero informado X 3 = " << R << endl << endl;
    cout << "Deseja continuar? (S/N): ";
    cin >> RESP;
    RESP = toupper(RESP);
    cout << endl;
  }
  return 0;
}

File: Cap 05/5.3 - Laco Condicional Pre-Teste/5.3.1 - Controle Condicional Verdadeiro/Efetivo/Interativo/enquantoINT.ts
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 * 
 * Capitulo ..: 5 - Topico 5.3.1
 * 
 * Programa ..: LACO CONDICIONAL PRE-TESTE COM FLUXO VERDADEIRO INTERATIVO
 */

import readlineSync from 'readline-sync';

console.log('LACO PRE-TESTE COM FLUXO VERDADEIRO');
console.log('Multiplicacao de numero por 3 por N vezes');
console.log();
let RESP: string = 'S';
while (RESP === 'S') {
  const N: number = parseInt(readlineSync.question('Entre valor numerico inteiro: '));
  const R: number = N * 3;
  console.log('O numero informado X 3 = ', R);
  console.log();
  RESP = readlineSync.question('Deseja continuar? (S/N): ').toUpperCase();
  console.log();
}
